### PR TITLE
(chore)Separated Compact Patient Search Extension from the original code

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR separates Compact Patient Search Extension from the original code and exportes the extension from Patient Search app. 

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Review by ZacButko — COMMENTED** [review]
I apologize @vasharma05 . I think we will need to find a better way to export this component. Currently it is being exported out of `esm-patient-search-app` which as @ibacher points out [here](https://github.com/openmrs/openmrs-esm-patient-chart/pull/667#issuecomment-1108699755) will create weird unwanted dependencies among **apps** . Is there a way that this could be exported from a new module not dependent on any app in this repo to avoid potential circular import errors? If not then exporting the component directly can wait until we have a more mature repo structure available to allow such a thing

File: packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx [hunk 1]
```diff
@@ -87,9 +87,9 @@ const PatientSearchResults: React.FC<PatientSearchResultsProps> = ({ patients, h
 
   return (
     <>
-      {fhirPatients.map((patient) => (
+      {fhirPatients.map((patient, indx) => (
         <ConfigurableLink
-          onClick={(evt) => onClickSearchResult(evt, patient.id)}
+          onClick={(evt) => onClickSearchResult(evt, patients[indx])}
```
Comment (ZacButko) [inline packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx:92]: awesome

File: packages/esm-appointments-app/translations/en.json [hunk 2]
```diff
@@ -14,31 +16,37 @@
   "appointmentReminder": "Would Patient like to get a reminder about appointment",
   "appointments": "Appointments",
   "appointmentScheduled": "Appointment scheduled",
+  "appointmentsList": "Appointment list is empty",
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-appointments-app/translations/en.json]: This key isn't added by me. It's added in the appointments, and came up when extracting the translations. I'll fix it.

File: packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx (hunk 1 above)
Comment (vasharma05) [reply packages/esm-patient-search-app/src/compact-patient-search/compact-patient-banner.component.tsx:92]: Thanks!

**Comment by vasharma05** [comment]
> I apologize @vasharma05 . I think we will need to find a better way to export this component. Currently it is being exported out of esm-patient-search-app which as @ibacher points out https://github.com/openmrs/openmrs-esm-patient-chart/pull/667#issuecomment-1108699755 will create weird unwanted dependencies among apps . Is there a way that this could be exported from a new module not dependent on any app in this repo to avoid potential circular import errors? If not then exporting the component directly can wait until we have a more mature repo structure available to allow such a thing

No issues @ZacButko!
I'll delete the export and I totally agree with Ian here. But can we still merge this PR?
The reason being, now I believe the extension moved into a different folder is a better approach. I'll just push the new commit removing the export.
Thanks!

**Comment by ZacButko** [comment]
Yes, I look forward to the small fixes, and then we can merge.

**Comment by vasharma05** [comment]
Hey @ZacButko!
This PR is ready to be merged.

File: packages/esm-patient-search-app/src/compact-patient-search-extension/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+const CompactPatientSearchComponent: React.FC<CompactPatientSearchProps> = ({
+  selectPatientAction,
+  initialSearchTerm = '',
+  buttonProps,
+}) => {
+  const [searchTerm, setSearchTerm] = useState(initialSearchTerm);
+  const handleSearchTerm = debounce((val) => setSearchTerm(val), 300);
+  const showSearchResults = useMemo(() => !!searchTerm.trim(), [searchTerm]);
```
Comment (ZacButko) [inline packages/esm-patient-search-app/src/compact-patient-search-extension/index.tsx:21]: I think this unprotected trim is causing errors for me. Can you add an optional chaining here?

**Review by ZacButko — APPROVED** [review]
Looks good. I'm still a little worried that I'm seeing errors from `searchTerm.trim()` from this [commit](https://github.com/openmrs/openmrs-esm-patient-management/commit/6598a5c3db100a4df2f7f066ce8c9a1244c1d271#diff-5ee6c377cc442fdae47950f91319504ac5c8ed66fb8acb48d71a7c9f10bb2159R28). Can you take one more look to make sure?

File: packages/esm-patient-search-app/src/compact-patient-search-extension/index.tsx (hunk 3 above)
Comment (vasharma05) [reply packages/esm-patient-search-app/src/compact-patient-search-extension/index.tsx:21]: Actually the prop `initialSearchTerm` was undefined, that caused the error. I gave it a default value in here, and it works as needed. I'll add an optional chaining. No issues.
Thanks!

**Pull request merged**